rfcs/2021-03-02-6600-vector-remap-language-web.md

```diff
@@ -0,0 +1,169 @@
+# RFC 6600 - 2021-03-03 - Vector Remap Language on the web
+
+From the beginning of our work on [Vector Remap Language][VRL] (VRL), we've had discussions about
+novel ways to make VRL approachable and even downright fun for users. Thus far, we've done some
+pretty unique things, like providing a REPL for the language. But we've also had plans, though
+fairly unspecific ones, about making VRL available on the web in some form. This RFC seeks to lend
+some concreteness to those discussions.
+
+## Options
+
+In this RFC, I discuss three possibities. The first two emerged from our internal discussions, while
+the third was gathered from a survey of the Wasm ecosystem.
+
+1. An in-browser [web app][web_app]/playground for experimenting with and learning VRL
+2. A [web server][web_server] that can process VRL programs and events
+3. A [published Wasm module], largely for internal use
+
+I propose pursuing #1 and #3 in the near term, while leaving #2 as an option for the future.
+
+### Interactive web app
+
+An interactive VRL web app would be a browser-based app that provides an aesthetically pleasing and
+UX-friendly experience for learning and experimenting with VRL. In its most basic form, it would
+enable you to enter a VRL event and a VRL program into a text area and see what happens. But there
+are other capabilities that we should consider:
+
+* A step-by-step tutorial, like the one under way in PR [6184] that provides real exercises with
+  right and wrong answers
+* A built-in "library" of example events (JSON, Syslog, AWS formats, etc.)
+* The ability to share URLs for specific VRL events and/or programs. This would likely require using
```

Comment:
I think the ability to share is an important one to making this successful but it ups the complexity of the project.  If we want to support that I think we have to support versions so an old link will continue to work.  I think the TS playground is an example where the inputs are encoded in the URL so we don't need to store this in a backend somewhere.  https://www.typescriptlang.org/play